app/models/publisher.rb

```diff
@@ -19,6 +19,27 @@ def publish
     end
   end
 
+  def discard_draft
+    begin
+      ActiveRecord::Base.transaction do
+        publishing_api.discard_draft(content_model.content_id)
+
+        latest_published_edition = content_model.editions.published.last
+        if latest_published_edition.present?
+          content_model
+            .editions
+            .where("created_at > ?", latest_published_edition.created_at)
+            .destroy_all
+        else
+          content_model.destroy!
```

Comment:
It's possible to add a guide to a topic that hasn't been published and I believe the content team are doing it a lot as they prepare sections to publish all at once. I think the chances of it happening are fairly decent which would leave the data in a broken state.